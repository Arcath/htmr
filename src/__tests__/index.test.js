/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import toReactElement from '../';

test('convert correctly', () => {
  const html = '<p id="test">This is cool</p>';
  const content = toReactElement(html);

  const tree = renderer.create(content);
  expect(tree).toMatchSnapshot();
});

test('self closing component', () => {
  const html = `
    <div>
      <img src="https://www.google.com/logo.png" />
      <iframe src="https://www.youtube.com/embed/I2-_iLzmkVw"></iframe>
    </div>
  `;

  const content = toReactElement(html);
  const tree = renderer.create(content);
  expect(tree).toMatchSnapshot();
});

test('custom component', () => {
  const html = '<p data-custom="true">Custom component</p>';

  const Paragraph = ({ children, ...props }) =>
    <p {...props} className="css-x243s">
      {children}
    </p>;

  const nodeMap = { p: Paragraph };
  const content = toReactElement(html, nodeMap);
  const tree = renderer.create(content);

  expect(tree).toMatchSnapshot();
});

test('multi children', () => {
  const html = `
    <p style="margin: 0px">Multi</p>
    <p style="display: none">Component</p>
  `;
  const content = toReactElement(html);

  const tree = renderer.create(
    <div className="wrapper">
      {content}
    </div>
  );

  expect(tree).toMatchSnapshot();
});
