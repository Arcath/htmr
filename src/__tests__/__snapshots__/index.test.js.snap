// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert correctly 1`] = `
<p
  id="test"
>
  This is cool
</p>
`;

exports[`custom component 1`] = `
<p
  className="css-x243s"
  data-custom="true"
>
  Custom component
</p>
`;

exports[`multi children 1`] = `
<div
  className="wrapper"
>
  <p
    style={
      Object {
        "margin": "0",
      }
    }
  >
    Multi
  </p>
  <p
    style={
      Object {
        "display": "'none'",
      }
    }
  >
    Component
  </p>
</div>
`;

exports[`self closing component 1`] = `
<div>
  <img
    src="https://www.google.com/logo.png"
  />
  <iframe
    src="https://www.youtube.com/embed/I2-_iLzmkVw"
  />
</div>
`;
